File: Leetcode1009. Complement of Base 10 Integer/Solution.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) return 1;
        int bits = 32 - Integer.numberOfLeadingZeros(n);
        int mask = (1 << bits) - 1;
        return ~n & mask;
    }
}


// bit smearing
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) return 1;
        int mask = n;
        for (int i = 0; i <= 4; i++)
            mask |= mask >> (1 << i);
        return ~n & mask;
    }
}

File: Leetcode1009. Complement of Base 10 Integer/Solution.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0: return 1
        mask = n
        for i in (1, 2, 4, 8, 16):
            mask |= mask >> i
        return ~n & mask


class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0: return 1
        return ~n & (1 << n.bit_length()) - 1
